nixos/lib/test-driver/test-driver.py

```diff
@@ -441,7 +441,7 @@ def require_unit_state(self, unit: str, require_state: str = "active") -> None:
     def execute(self, command: str) -> Tuple[int, str]:
         self.connect()
 
-        out_command = "( {} ); echo '|!=EOF' $?\n".format(command)
+        out_command = "( set -o pipefail; {} ); echo '|!=EOF' $?\n".format(command)
```

Comment:
```suggestion
        out_command = "( set -euo pipefail; {} ); echo '|!=EOF' $?\n".format(command)
```

Besides `false | echo ok`, I would expect `false; echo ok` and `echo $undefined` to fail.
Optional variables (unlikely in this use case) can still be used via `${undefined:-}` if necessary.

Reply:
Yeah, I agree. I've pushed the change and updated the manual.